src/compiler/crystal/types.cr

```diff
@@ -312,6 +312,15 @@ module Crystal
       nil
     end
 
+    def superclasses
+      superclasses = [] of Type
+      klass = self
+      while klass = klass.superclass
+        superclasses << klass
+      end
+      superclasses
+    end
```

Comment:
We already have `superclass`, I'm not sure we need `superclasses` —which doesn't exist in Ruby, by the way. Both for the compiler (not used) and for `@type`. We can already iterate from parent to parent if needed.